packages/ui/src/editable-textarea/EditableTextarea.tsx

```diff
@@ -0,0 +1,139 @@
+"use client";
+import {
+  Editable,
+  type EditableRootProps,
+  useEditableContext,
+} from "@ark-ui/react";
+import { PencilIcon } from "@heroicons/react/24/outline";
+import { cva } from "class-variance-authority";
+import { type ElementRef, forwardRef } from "react";
+import { Button } from "../button";
+import { Textarea } from "../textarea";
+import type { ValueOrFunction } from "../types";
+import { classNames, getValue } from "../utils";
+
+export type EditableTextarea = Omit<EditableRootProps, "activationMode"> & {
+  isReadOnly?: ValueOrFunction<boolean>;
+  isDisabled?: ValueOrFunction<boolean>;
+  isLoading?: ValueOrFunction<boolean>;
+  size?: "sm" | "md" | "lg";
+};
+
+export const EditableTextarea = forwardRef<
+  ElementRef<typeof Editable.Root>,
+  EditableTextarea
+>(
+  (
+    {
+      size = "md",
+      className,
+      isDisabled = false,
+      isReadOnly = false,
+      isLoading = false,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const disabled =
+      props.disabled || getValue(isDisabled) || getValue(isLoading);
+    const readOnly = props.readOnly || getValue(isReadOnly);
+
+    return (
+      <Editable.Root
+        {...props}
+        activationMode="dblclick"
+        readOnly={readOnly}
+        disabled={disabled}
+        className={classNames("w-full", className)}
+        ref={forwardedRef}
+      >
+        <Editable.Context>
+          {() => (
+            <EditableItem size={size} readOnly={readOnly} disabled={disabled} />
+          )}
+        </Editable.Context>
+      </Editable.Root>
+    );
+  },
+);
+
+EditableTextarea.displayName = "EditableTextarea";
+
+const PREVIEW_CLASS = {
```

Comment:
rename this variable to editableTextareaPreviewClasses and change this object just like I did in editable number and editable text.